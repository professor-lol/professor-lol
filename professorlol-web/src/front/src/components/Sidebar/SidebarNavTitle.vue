<template>
  <li :class="classList">
    <template v-if="wrapper && wrapper.element">
      <component v-bind:is="wrapper.element" v-bind="wrapper.attributes">
        {{name}}
      </component>
    </template>
    <template v-else>
      {{name}}
    </template>
  </li>
</template>

<script>
export default {
  props: {
    name: {
      type: String,
      default: ''
    },
    classes: {
      type: [String, Array, Object],
      default: ''
    },
    wrapper: {
      type: Object,
      default: () => {}
    }
  },
  computed: {
    classList () {
      return [
        'nav-title',
        ...this.itemClasses
      ]
    },
    itemClasses () {
      const classes = this.classes
      return !classes ? [] : typeof classes === 'string' || classes instanceof String ? classes.split(' ') : Array.isArray(classes) ? classes : Object.keys(classes).filter(i=>classes[i])
    }
  }
}
</script>
