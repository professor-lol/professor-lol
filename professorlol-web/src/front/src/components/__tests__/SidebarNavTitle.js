import { mount } from "@vue/test-utils";
import SidebarNavTitle from "../Sidebar/SidebarNavTitle";

describe("SidebarNavTitle.vue", () => {
  // Inspect the raw component options
  it("should have default props", () => {
    const wrapper = mount(SidebarNavTitle, {
      propsData: {
        name: 'test',
        classes: '',
        wrapper: {}
      }
    })
    expect(wrapper.props().name).toEqual('test')
    expect(wrapper.props().classes).toBe('')
    expect(typeof wrapper.props().wrapper).toBe('object')
  });
  it('has classList computed property', () => {
    expect(typeof SidebarNavTitle.computed.classList).toBe('function')
  })
  it('has itemClasses computed property', () => {
    expect(typeof SidebarNavTitle.computed.itemClasses).toBe('function')
  })
  it('renders correctly', () => {
    const wrapper = mount(SidebarNavTitle)
    expect(wrapper.element).toMatchSnapshot()
    expect(wrapper.classes()).toContain('nav-title')
  })
});
