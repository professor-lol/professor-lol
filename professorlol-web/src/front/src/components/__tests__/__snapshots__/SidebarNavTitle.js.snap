// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarNavTitle.vue renders correctly 1`] = `
<li
  class="nav-title"
>
  
    
  
</li>
`;
